### ISSUE TITLE
Oracle: Default value from sequence causes file parsing to fail

### ISSUE THREAD

**Issue by saulotoledo** [sqlfluff/sqlfluff#7038]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The parsing of the file in the "how to reproduce" section fails when we have a default value coming from a sequence.
In addition to that, even when the parsing works, there is always a message `[1 unfixable linting violations found]` and the process exits with error code 1.

### Expected Behaviour

The file parsing should work for all rules. The message at the bottom should either display `[0 unfixable linting violations found]` or not appear at all. The process should exit successfully with exit code 0.

### Observed Behaviour

– File parsing fails.
– The message `[1 unfixable linting violations found]` appears even when no errors are present.
– The process always exits with error code 1.

### How to reproduce

1. Create a migration file (e.g., `/workspace/migrations/example_migration.sql`) as follows:

```sql
CreaTE SEQUENCE seq_mY_TabLe START WITH 1;

create TaBlE mY_Table 
(
  id NUMBER(18) DEFAULT seq_mY_TabLe.NEXTVAL,
  name VARCHAR2(100),
  description VARCHAR2(255),
  created_at timestamp DEFAULT current_TIMESTAMP,
  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
  
  CONStrAINT my_table_pk PRIMARY KEY (id)
);
```

2. Configure sqlfluff as needed (see configuration section).
3. Install and run sqlfluff on that file:

```bash
pip install --no-cache-dir sqlfluff==3.4.2
sqlfluff fix /workspace/migrations/example_migration.sql
```

The response from sqlfluff is below. Note that the file parsing fails:

```bash
==== finding fixable violations ====
== [/workspace/migrations/example_migration.sql] FAIL
L:   1 | P:   1 | CP01 | Keywords must be upper case. [capitalisation.keywords]
L:   1 | P:  17 | CP02 | Unquoted identifiers must be lower case.
                       | [capitalisation.identifiers]
L:   3 | P:   1 | CP01 | Keywords must be upper case. [capitalisation.keywords]
L:   3 | P:   8 | CP01 | Keywords must be upper case. [capitalisation.keywords]
L:   3 | P:  14 | CP02 | Unquoted identifiers must be lower case.
                       | [capitalisation.identifiers]
L:   8 | P:  14 | CP05 | Datatypes must be upper case. [capitalisation.types]
L:   8 | P:  32 | CP03 | Function names must be upper case.
                       | [capitalisation.functions]
L:  11 | P:   3 | CP01 | Keywords must be upper case. [capitalisation.keywords]
== [/workspace/migrations/example_migration.sql] FIXED
8 fixable linting violations found
  [1 unfixable linting violations found]
```

4. Replace `DEFAULT seq_mY_TabLe.NEXTVAL` with `DEFAULT 1` and run sqlfluff twice:

```bash
sqlfluff fix /workspace/migrations/example_migration.sql
sqlfluff fix /workspace/migrations/example_migration.sql
```

The first run fixes the file, but the second still displays `[1 unfixable linting violations found]`, and the process exits with code 1.

```bash
| ==== finding fixable violations ====
| ==== no fixable linting violations found ====
| All Finished!
|   [1 unfixable linting violations found]
```

### Dialect

oracle

### Version

3.4.2

### Configuration

I tested it with the following `.sqlfluff` configuration file in `/workspace`:

```ini
[sqlfluff]
templater = jinja
dialect = oracle
ignore = templating
large_file_skip_byte_limit = 0
max_line_length = 0

[sqlfluff:layout:type:alias_expression]
spacing_before = align
align_within = select_clause
spacing_after = touch

[sqlfluff:indentation]
tab_space_size = 2
indent_unit = space
indented_joins = false
indented_using_on = true
allow_implicit_indents = true
indented_on_contents = false
indented_ctes = false

[sqlfluff:rules:aliasing.table]
aliasing.table = explicit

[sqlfluff:rules:aliasing.column]
aliasing.column = explicit

[sqlfluff:rules:aliasing.expression]
allow_scalar = True

[sqlfluff:rules:ambiguous.join]
fully_qualify_join_types = inner

[sqlfluff:rules:ambiguous.column_references]
group_by_and_order_by_style = consistent

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower
unquoted_identifiers_policy = all

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:jinja.padding]
single_space = true

[sqlfluff:rules:layout.spacing]
no_trailing_whitespace = true
extra_whitespace = false

[sqlfluff:rules:layout.commas]
line_position = trailing

[sqlfluff:rules:layout.functions]
no_space_after_function_name = true

[sqlfluff:rules:layout.select_targets]
wildcard_policy = single

[sqlfluff:rules:layout.set_operators]
set_operator_on_new_line = ['UNION', 'UNION ALL']

[sqlfluff:rules:structure.nested_case]

[sqlfluff:rules:structure.subquery]
forbid_subquery_in = join
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
